[WIP] Add span rendering template and clean up span briefs and notes (#1505)
open-telemetry/semantic-conventions: docs/gen-ai/gen-ai-spans.md

<!-- markdownlint-capture -->
<!-- markdownlint-disable -->

**Status:** ![Development](https://img.shields.io/badge/-development-blue)

> This span represents client call to Generative AI model or service.

Reviewer: ```suggestion
This span represents a client call to Generative AI model or service.
```

**Span name** SHOULD be `{gen_ai.operation.name} {gen_ai.request.model}`.
Semantic conventions for individual GenAI systems and frameworks MAY specify different span name format
and MUST follow the overall [guidelines for span names](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.37.0/specification/trace/api.md#span).

**Span kind** SHOULD be `CLIENT`and MAY be set to `INTERNAL` on spans representing
call to models running in the same process. It's RECOMMENDED to use `CLIENT` kind

> when GenAI system being instrumented usually runs in a different process than its

Reviewer: ```suggestion
when the GenAI system being instrumented usually runs in a different process than its
```

> client or when GenAI call happens over instrumented protocol such as HTTP.

Reviewer: ```suggestion
client or when the GenAI call happens over instrumented protocol such as HTTP.
```

<!-- markdownlint-capture -->
<!-- markdownlint-disable -->

**Status:** ![Development](https://img.shields.io/badge/-development-blue)

Describes tool execution span.

`gen_ai.operation.name` SHOULD be `execute_tool`.

**Span name** SHOULD be `execute_tool {gen_ai.tool.name}`.

GenAI instrumentations that are able to instrument tool execution call SHOULD do so.
However, it's common for tools to be executed by the application code. It's recommended

> for the application developers to follow this semantic conventions for tool invoked

Reviewer: ```suggestion
for the application developers to follow this semantic convention for tools invoked
```

by the application code.

> **Span kind** SHOULD be `INTERNAL`.

Reviewer: are these out of proc tools?

Author: can be in-proc or out-of-proc

Author: It's essentially a function call which may involve a network one.

E.g. I have a `get_weather` tool that retrieves a weather from a remote service, but may also have a local cache. 
Or a tool that asks user for additional input.

What instrumentation has is the reference to the function to call and it can wrap the execution into the span, but it won't know what happens underneath.

Plus, the network operations inside the function call would be traced (if instrumented) and may have client (logical) -> client (network)  layers already. So I'd rather keep `internal` kind.

PS: I almost wish we had a `¯\_(ツ)_/¯` kind instead of `INTERNAL`

Reviewer: > the network operations inside the function call would be traced (if instrumented)

👍
